raster-foundry/blasterjs
# Dialog and button enhancements

## Overview

Button
- Theme structure changed so default colors are now configured in `colors`.

Dialog
- Overlay color and opacity can now be themed or propped.
- Custom transitions can now be themed.
- z-index can now be propped.
- Focus ring color can now be themed, and has a default value.

### Checklist


### Upgrade instructions

If there are any of the following in this PR, provide proper instructions on how to upgrade:


Button theme changes:
- `intents.colors.default` is now `base.colors.color`
- `base.colors.defaultBg` is now `base.colors.bg`
- `base.colors.defaultBorder` is now `base.colors.border`
- `base.colors.darkText` is now `base.colors.colorDark`
- `base.colors.lightText` is now `base.colors.colorLight`
- `base.hover.colors.defaultBg` is now `base.colors.bgHover`
- All other `base.hover.*` properties deprecated, and `base.hover` removed.

Dialog prop changes:
- `opacity` should now affect dialog opacity, not overlay opacity
- Added `overlayOpacity`
- Added `overlayColor`
- Added `zIndex`
- Deprecated: `portalClassName`, `overlayClassName`, `className`, `bodyOpenClassName`, `htmlOpenClassName`

Dialog theme changes:
- Added `colors.overlay`
- Added `colors.focus`
- Added `zIndices.zIndex`
- Added `opacities.overlay`
- Added `durations.closeTimeoutMS`
- Added `transitions.overlay.{base, afterOpen, beforeClose}`
- Added `transitions.dialog.{base, afterOpen, beforeClose}`

### Notes

Tweaked `replaceThemeRefs()` in utils.js to help prevent call stack errors.

Hard coded Dialog classnames, for simplicity.

Hid Field, ToggleField, and NavBar docs pages. They're to be deprecated.

## Testing Instructions

- Fiddle with new Button theme values. Confirm works as expected.
- Fiddle with new Dialog props and theme values. Confirm works as expected.

Closes #210 
Closes #211 
Closes #212 
Closes #213
